Update DG Appendix (#303)
AY2425S1-CS2103-F12-3/tp: docs/DeveloperGuide.md

      Expected: Similar to previous.

### Finding a restaurant by name

1. Find a restaurant by its saved name

   1. Prerequisites: The requested restaurant has to be currently saved.<br>
   The first test case below assumes the user has not added any more restaurants with the name Swenswen.

   1. Test case: `find swenswen`<br>
      Expected: 1 restaurant listed!. Details of the restaurant message is shown on the right hand side.

   1. Test case: `find !@#`<br>
      Expected: Message displays no restaurants found.

   1. Other incorrect find commands to try: `find`<br>
      Expected: Error details shown in the status message. Status bar remains the same.

### Filtering by tags

1. Find a restaurant by its tag names

   1. Prerequisites: The requested restaurant has to be currently saved.<br>
   The first test case below assumes the user has not added any more restaurants with the tag halal.

   1. Test case: `tags halal`<br>
      Expected: 3 restaurant listed!. Details of the restaurant message is shown on the right hand side.

   1. Test case: `tags $$$`<br>
      Expected: Message displays no restaurants found.

>    1. Other incorrect find commands to try: `tags`<br>
>       Expected: Error details shown in the status message. Status bar remains the same.

Reviewer: ```suggestion
   1. Other incorrect Filter-by-Tag commands to try: `tags`<br>
      Expected: Error details shown in the status message. Status bar remains the same.
```

### Filtering by price

1. Find a restaurant by its price labels

   1. Prerequisites: The requested restaurant has to be currently saved.<br>
      The first test case below assumes the user has not added any more restaurants with the price label of `$`.

>    1. Test case: `tags $`<br>
>       Expected: 3 restaurant listed!. Details of the restaurant message is shown on the right hand side.

Reviewer: ```suggestion
   1. Test case: `price $`<br>
      Expected: 3 restaurant listed!. Details of the restaurant message is shown on the right hand side.
```

   1. Test case: `price expensive`<br>
      Expected: Message displays no restaurants found. 

>    1. Other incorrect find commands to try: `price`<br>
>       Expected: Error details shown in the status message. Status bar remains the same.

Reviewer: ```suggestion
   1. Other incorrect Filter-by-Price commands to try: `price`<br>
      Expected: Error details shown in the status message. Status bar remains the same.
```

### Rating a restaurant

1. Rating a restaurant while all restaurants are being shown

1. Prerequisites: The requested restaurant has to be currently saved.<br>

    1. Test case: `rate 1 r/2`<br>
       Expected: First contact is given a rating of `2` from the list. Left hand side displays updated rating for the restaurant.

    1. Test case: `rate 0`<br>
       Expected: No restaurant is rated. Error details shown in the status message. Status bar remains the same.

>     1. Other incorrect delete commands to try: `rate`<br>
>        Expected: Similar to previous.

Reviewer: ```suggestion
    1. Other incorrect Rate commands to try: `rate`<br>
       Expected: Similar to previous.
```

### Favourite a restaurant

1. Favourite a restaurant while all restaurants are being shown

> 1. Prerequisites: The requested restaurant has to be currently saved.<br>
> 

Reviewer: ```suggestion
1. Prerequisites: The requested restaurant has to be currently saved.<br>
In the test case below, we assume that none of the restaurants in the list were previously marked as favourite. 
```

    1. Test case: `fav 2`<br>
       Expected: Second contact has its border highlighted orange and brought to the top of the list as the first index.

    1. Test case: `fav 0`<br>
       Expected: No restaurant is turned to favourite. Error details shown in the status message. Status bar remains the same.

    1. Other incorrect delete commands to try: `fav`<br>
       Expected: Similar to previous.

### Un-favourite a restaurant

1. Un-favourite a restaurant while all restaurants are being shown

1. Prerequisites: The requested restaurant has to be currently saved and is a favourite restaurant as shown above in the previous test.<br>

>     In the test case below, we assume that the first restaurant in the list is a favourite as described by the previous test case mentioned above.

Reviewer: ```suggestion
    In the test case below, we assume that the first restaurant in the list is a marked as favourite as described by the previous test case mentioned above.
```

    1. Test case: `unfav 1`<br>
       Expected: First contact has its border un-highlighted and brought back to its original index in the list before being set as a favourite.

    1. Test case: `unfav 0`<br>
       Expected: No favourite restaurant is turned back to normal. Error details shown in the status message. Status bar remains the same.

>     1. Other incorrect delete commands to try: `unfav`<br>
>        Expected: Similar to previous.

Reviewer: ```suggestion
    1. Other incorrect Un-favourite commands to try: `unfav`<br>
       Expected: Similar to previous.
```

> ## **Appendix: Effort**

Reviewer: Looks good!

Luckily the diagrams that were provided indeed helped a lot in our teams journey in understanding and developing the project.
One of the methods that helped us a lot in our speed and ability to meet all the necessary datelines were to take inspiration of some of the code
that had already been implemented. For example, the source code for the `tags` command and feature was heavily inspired by the 
already existing `find` command and feature that came with the project.<br>

Additionally, another tough lesson we learnt was to not be too over ambitious in our planned features and implementation. At the start of the project, 
while we were ambitious with excitement on what were the list of features we wanted to add, we realised that scrapping some ideas we created was the best 
strategy as at the end of the day, we had to deliver iterations within a one-week time span, thus, looking back, while unfortunate, it was the most sound strategy we could 
have adopted, which was to work on features that we could deliver by the stipulated dateline.

## **Appendix: Planned Enhancements**
Team size: 4<br>

> 1. **Make add command accept non-alphanumeric values:** The current implementation for the add command is such where names and tags can only be saved as alphanumeric values. We plan to allow adding non-alphanumeric

Reviewer: ```suggestion
1. **Enhance `add` command to accept non-alphanumeric values:** The current implementation for the add command is such where names and tags can only be saved as alphanumeric values. We plan to allow adding non-alphanumeric
```

values for names and tags in the future. E.g. `add n/L'Éclat Culinaire p/12345678 e/test@email.com a/Pasir Ris pr/$$$ t/Crème brûlée`.
2. **Allow spaces and special characters within phone numbers:** The current implementation of adding contacts with phone numbers does not allow spaces and only allows numbers. It would be more practical if
spaces and special characters are allowed for longer phone numbers with specific area and country codes. For example: `12345678` is what can be accepted currently. We plan on allowing phone numbers like: `+65 1234 5678`.

> 3. **Fix grammar for 1 restaurant listed:** Currently the string that is returned when 1 restaurant is found in any filter command is as follows: `1 restaurants listed!`, which is grammatically incorrect. We plan on 

Reviewer: ```suggestion
3. **Correct grammar for "1 restaurants listed" message:** Currently the string that is returned when 1 restaurant is found in any filter command is as follows: `1 restaurants listed!`, which is grammatically incorrect. We plan on 
```

fixing this in the future to: `1 restaurant listed!`

> 4. **Improve tags error message:** Currently tags are only allowed to be alphanumeric values, if the user were to search `tags $$$$`, the response of `0 restaurants listed!` is the output. However, a better implementation would have been 

Reviewer: ```suggestion
4. **Improve `tags` error message:** Currently tags are only allowed to be alphanumeric values, if the user were to search `tags $$$$`, the response of `0 restaurants listed!` is the output. However, a better implementation would have been 
```

for the source code to first check if the keywords fits the current criteria for what is allowed as tags before finding. If the conditions were not met, a more appropriate response could have been displayed, one example 
that would be a response to the above example, would be `tags should only be alphanumeric values, if you are trying to filter by price use to price command`.

> 5. **Improve price error message:** Currently price are only allowed to be filtered by a given combination of the allowed price labels, if the user were to search `price $$$$$` or `price 4`, the response of `0 restaurants listed!` is the output. However, like the previous point, 

Reviewer: ```suggestion
5. **Improve `price` error message:** Currently price are only allowed to be filtered by a given combination of the allowed price labels, if the user were to search `price $$$$$` or `price 4`, the response of `0 restaurants listed!` is the output. However, like the previous point, 
```

a better implementation would have been for the source code to first check if the input were price labels before finding. If the conditions were not met, a more appropriate response could have been displayed, for example 
`Price command only accepts a combination of the following: $ $$ $$$ $$$$`.

> 6. **Improve edit command for checking phone numbers:** Currently a user can enter a really long phone number, e.g. `120 digits` when editing an already existing contact with the `edit` command. We plan on checking the user inputs in the future for the edit command so as to prevent such 

Reviewer: ```suggestion
6. **Limit phone number length in `edit` command:** Currently a user can enter a really long phone number, e.g. `120 digits` when editing an already existing contact with the `edit` command. We plan on checking the user inputs in the future for the edit command so as to prevent such 
```
